projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.ts

```diff
@@ -588,9 +604,27 @@ export class IgxHierarchicalGridComponent extends IgxHierarchicalGridBaseCompone
 
     private hg_verticalScrollHandler(event) {
         this.scrollTop = event.target.scrollTop;
+        this.hideOverlays();
+    }
+
+    public onContainerScroll() {
+        this.hideOverlays();
     }
 
     private hg_horizontalScrollHandler(event) {
         this.scrollLeft = event.target.scrollLeft;
+        this.hideOverlays();
+    }
+
+    private hideOverlays() {
+        this._overlayIDs.forEach(overlayID => {
+            this.overlayService.hide(overlayID);
+            // blur in case some editor somewhere decides to move focus back
+            this.overlayService.onClosed.pipe(
+                filter(o => o.id === overlayID),
+                takeUntil(this.destroy$)).subscribe(() => {
+                    (document.activeElement as any).blur();
```

Comment:
@kdinev , atm we don't have a better solution for this as it seems to be caused by the for-of optimization with moving views around. However, the ViewRef that it works with has no public properties that will allow us to target blur detached elements.

This fix has better performance than anything that can be done in for-of.

Reply:
Perhaps focusing the H Grid (what happens right now when you scroll normally) instead of random element blur would be better and should fail gracefully if the grid is already focused.
This, of course, assumes that the grid keeps its `tabindex` and remains focusable.

Reply:
@damyanpetev Agreed.